File: 卡码/代码随想录/1/09-动态规划/Js/02-背包问题/01-01背包/LeetCode_0474.js
//474. 一和零
const findMaxForm = function(strs, m, n) {
	const dp=new Array(m+1).fill(0).map(item=>new Array(n+1).fill(0));
	for(let str of strs){
		let x=0,y=0;
		for(let ch of str){
			if(ch=="0") x++;
			else y++;
		}
		for(let i=m;i>=x;i--){
			for(let j=n;j>=y;j--){
				dp[i][j]=Math.max(dp[i][j],dp[i-x][j-y]+1);
			}
		}
	}
	return dp[m][n];
};

File: 卡码/代码随想录/1/09-动态规划/Java/dp/src/main/java/com/kurumi/dp/_02_背包问题/_01_01背包/LeetCode_0474.java
//474. 一和零
package com.kurumi.dp._02_背包问题._01_01背包;

public class LeetCode_0474 {

}
class Solution {
  public int findMaxForm(String[] strs, int m, int n) {
    int[][] dp=new int[m+1][n+1];
    for(String str:strs){
      int x=0,y=0;
      for(int i=0;i<str.length();i++){
        if(str.charAt(i)=='0') x++;
        else y++;
      }
      for(int i=m;i>=x;i--){
        for(int j=n;j>=y;j--){
          dp[i][j]=Math.max(dp[i][j], dp[i-x][j-y]+1);
        }
      }
    }
    return dp[m][n];
  }
}
